File: 프로그래머스/나누어떨어지는숫자배열.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
	vector<int> answer;

	int cnt = 0;
	for (int i = 0; i < arr.size(); i++) {
		if ((arr[i] % divisor) == 0) {
			answer.push_back(arr[i]);
			cnt++;
		}
	}
	if (cnt == 0) {
		answer.push_back(-1);
	}

	sort(answer.begin(), answer.end());
	return answer;
}

int main() {
	vector<int> array = { 5, 9, 7, 10 };
	int div = 5;
	cout << solution(array, div) << endl;

	return 0;
}

File: 프로그래머스/나누어떨어지는숫자배열.py
def solution(arr, divisor):
    answer = []
    for i in arr:
        if i % divisor == 0:
            answer.append(i)
    if len(answer) == 0:
        answer.append(-1)
    answer.sort()

    return answer


tarr = [5, 9, 7, 10]
tdivisor = 5

print(solution(tarr, tdivisor))
